code/modules/html_interface/map/station_map.dm

```diff
@@ -75,8 +92,68 @@
 			var/turf/tile = locate(i, r, map.zCentcomm)
 			if(tile && (is_type_in_list(tile.loc, allowed_areas) && !is_type_in_list(tile.loc, restricted_areas)))
 				if((!istype(tile, /turf/space) && istype(tile.loc, /area/mine/unexplored)) || istype(tile, /turf/simulated/wall) || istype(tile, /turf/unsimulated/mineral) || istype(tile, /turf/unsimulated/wall) || (locate(/obj/structure/grille) in tile) || (locate(/obj/structure/window/full) in tile) || istype(tile, /turf/simulated/shuttle/wall))
-					canvas.DrawBox("#FFFFFFDD", i, r)
+					if(map.holomap_offset_x.len >= map.zCentcomm)
```

Comment:
you mean the proc above? One draws the entire zLevel while the other only draws certain areas defined by the above filters.